astropy/units/tests/test_quantity_ufuncs.py

```diff
@@ -1003,6 +1004,29 @@ def test_ndarray_inplace_op_with_quantity(self):
         assert_array_equal(a, np.array([0.125, 1.25, 2.0]))
 
 
+class TestWhere:
+    """Test the where argument in ufuncs."""
+
+    @classmethod
+    def setup_class(self):
+        self.q = np.arange(4.0) << u.m
+
+    def test_where(self):
+        out = np.zeros_like(self.q)
+        result = np.add(self.q, 1 * u.km, out=out, where=[True, True, True, False])
+        assert result is out
+        assert_array_equal(result, [1000.0, 1001.0, 1002.0, 0.0] << u.m)
+
+    @pytest.mark.xfail(
+        NUMPY_LT_1_25, reason="where array_ufunc support introduced in numpy 1.25"
+    )
+    def test_exception_with_where_quantity(self):
+        a = np.ones(2)
+        where = np.ones(2, bool) << u.m
+        with pytest.raises(TypeError):
```

Comment:
should we use the `match=` keyword here?
```suggestion
        with pytest.raises(TypeError, match=r"Cannot cast array data from .* to .* according to the rule .*"):
```

Reply:
Yes, good idea, though I get a different error on numpy-dev (which I'll use...):
```
TypeError: operand type(s) all returned NotImplemented from __array_ufunc__(<ufunc 'add'>, '__call__', array([1., 1.]), array([1., 1.]), out=(array([1., 1.]),), where=<Quantity [ True,  True] m>): 'ndarray', 'ndarray', 'ndarray'
```

Reply:
Do the different messages have anything in common that we can use some regex? If not, I guess we don't have to include "match".